esp-hal/src/spi/mod.rs

```diff
@@ -64,7 +64,7 @@ impl embedded_hal::spi::Error for Error {
 /// and shifted.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
-pub enum SpiMode {
+pub enum Mode {
```

Comment:
This now makes the variants to be prefixed with the enum name. I'm also not entirely happy that this conflicts with `crate::Mode`.

Reply:
`embassy-stm32` and `embedded-hal` use a struct with `Polarity` and `Phase` (and call it just `Mode` including the name conflict for `embassy-stm32`)

We could name the variants s.th. like `ClockIdleLowCaptureOnFirstTransition` or `Cpol0Cpha0` but both is worse IMHO

Reply:
yikes, `CaptureOnFirstTransition` is obvious enough but when a datasheet calls out "This device can operate in SPI modes 1 and 3" it's also terribly unhelpful 😅

I can live with `Mode`, but I'd like to prevent triggering https://rust-lang.github.io/rust-clippy/master/index.html#enum_variant_names with the variant names.

Reply:
Renamed `crate::Mode` to `crate::DriverMode`, so this is no longer an issue

Reply:
DriverMode is no longer an issue, but `Mode::Mode0` is

Reply:
I guess we need to decide on https://github.com/esp-rs/esp-hal/issues/2716 to proceed

Reply:
We could use descriptive names for the enum variants (e.g. `CaptureOnFirstTransition` or something shorter) and define constants for `MODE_0` .. `MODE_3`